clang/utils/generate_unsupported_in_drivermode.py

```diff
@@ -0,0 +1,534 @@
+        f"{test_prefix}NOTE: To make changes, see llvm-project/clang/utils/generate_unsupported_in_drivermode.py"
+        + " from which it was generated.\n"
+        f"{test_prefix}NOTE: Regenerate this Lit test with the following:\n"
+        f"{test_prefix}NOTE: python generate_unsupported_in_drivermode.py "
+        "--options-td-dir llvm-project/clang/include/clang/Driver --llvm-include-dir llvm-project/llvm/include --llvm-tblgen llvm-project/build/bin/llvm-tblgen\n\n"
+    )
+
+
+def write_lit_test(test_path, test_visibility):
+    """Write the Lit tests to file
+    test_path: File write path
+    test_visibility: VISIBILITY_DEFAULT, VISIBILITY_FLANG, or VISIBILITY_FC1 which indicates whether to write
+    to the main Lit test file, the flang test file, or the flang -fc1 test file
+    """
+    lit_file = open(test_path, "w")
+
+    lit_file.write(get_lit_test_note(test_visibility))
+    batch_size = 100
+
+    for visibility, driver_data in driver_data_dict.items():
+        is_flang_pair = visibility == VISIBILITY_FLANG or visibility == VISIBILITY_FC1
+
+        if (
+            (test_visibility == VISIBILITY_FLANG and visibility != VISIBILITY_FLANG)
+            or (test_visibility == VISIBILITY_FC1 and visibility != VISIBILITY_FC1)
+            or (test_visibility == VISIBILITY_DEFAULT and is_flang_pair)
+        ):
+            continue
+
+        comment_str = EXCLAMATION if is_flang_pair else SLASH_SLASH
+
+        unflattened_option_data = list(
+            itertools.batched(driver_data.test_option_sequence, batch_size)
+        )
+
+        for i, batch in enumerate(unflattened_option_data):
+            # Example run line: "// RUN: not %clang -cc1 -A ... -x c++ - < /dev/null 2>&1 | FileCheck -check-prefix=CC1OptionCHECK0 %s"
+            run_cmd = (
+                f"\n{comment_str}RUN: not " + driver_data.lit_cmd_prefix
+            )  # "// RUN: not %clang -cc1 "
+
+            # // RUN: <command up to this point> \
+            # // RUN:   --one-option \
+            # // RUN:   -a-different-option \
+            # ...
+            run_cmd += f" \\\n{comment_str}RUN:   ".join(itertools.chain(("",), batch))
+
+            run_cmd += (
+                driver_data.lit_cmd_options  # "-x c++"
+                + driver_data.lit_cmd_end  # " - < /dev/null 2>&1 | FileCheck -check-prefix=CC1OptionCHECK"
+                + str(i)  # "0"
+                + " %s\n\n"  # " %s"
+            )
+
+            lit_file.write(run_cmd)
+
+            for option_str in batch:
+                # Example check line: "// CC1OptionCHECK0: {{(unknown argument).*-A}}"
+                check_cmd = (
+                    comment_str  # "//
+                    + visibility  # "CC1Option"
+                    + "CHECK"
+                    + str(i)  # "0"
+                    + ": {{("
+                    + driver_data.check_str  # "unknown argument"
+                    + ").*"
+                    + option_str.replace("+", "\\+")  # "-A"
```

Comment:
```suggestion
                    + re.escape(option_str)  # "-A"
```
This really should be minimally [re.escape](https://docs.python.org/3/library/re.html#re.escape), but it could be taken out of the regex too i.e.
`unknown argument{{.*}}-A`